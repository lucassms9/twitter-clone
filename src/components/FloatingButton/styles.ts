import styled from 'styled-components/native';
import FeatherIcon from 'react-native-vector-icons/Feather';

export const Container = styled.TouchableOpacity`
  width: 70px;
  height: 70px;
  border-radius: 35px;
  background-color: ${({ theme }) => theme.colors.button.active};
  position: absolute;
  bottom: 25px;
  right: 25px;
  justify-content: center;
  align-items: center;
`;

export const Icon = styled(FeatherIcon)`
  color: ${({ theme }) => theme.colors.text.primary};
`;
