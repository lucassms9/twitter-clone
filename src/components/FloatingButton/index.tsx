import React from 'react';

import { Container, Icon } from './styles';
import { Props } from './types';
const FloatingButton = ({ onPress }: Props) => {
  return (
    <Container onPress={onPress}>
      <Icon name='edit' size={25} />
    </Container>
  );
};

export default FloatingButton;
