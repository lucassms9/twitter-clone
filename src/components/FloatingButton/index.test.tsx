import React from 'react';

import { render } from '@tests/providers';

import FloatingButton from './index';

jest.mock('react-native-vector-icons/Feather', () => 'Icon');

describe('Floating Button component', () => {
  it('should render with success', () => {
    const { toJSON } = render(<FloatingButton onPress={jest.fn()} />, null);
    expect(toJSON()).toMatchSnapshot();
  });
});
