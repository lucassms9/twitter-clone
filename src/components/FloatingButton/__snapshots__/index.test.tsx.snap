// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floating Button component should render with success 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1d9bf0",
      "borderRadius": 35,
      "bottom": 25,
      "height": 70,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "right": 25,
      "width": 70,
    }
  }
  testID="FloatingButton"
>
  <Icon
    name="edit"
    size={25}
    style={
      Array [
        Object {
          "color": "#fff",
        },
      ]
    }
  />
</View>
`;
